### PR TITLE
Add PCOD Detection project - ML model for PCOS diagnosis with SHAP ex…

### LINKED ISSUE #1489: [Feature] PCOD Detection using Ensemble Learning, Explainable AI (XAI) & Counterfactual Learning
## Feature Request: PCOD Detection using Ensemble Learning, XAI & Counterfactual Learning

@Niketkumardheeryan , I want to work on this feature,
Current PCOD detection systems mainly focus on prediction accuracy but lack interpretability and transparency. In healthcare applications, black-box predictions can reduce trust among users, researchers, and medical professionals. There is also limited support for understanding *why* a prediction was made or *what changes* could alter the prediction outcome. This makes the system less actionable and less reliable for real-world medical assistance.

---

I would like to add a complete PCOD detection pipeline using Ensemble Learning combined with Explainable AI (XAI) and Counterfactual Learning.

The proposed solution should include:

* Data preprocessing and feature engineering
* Ensemble models such as:

  * Random Forest
  * XGBoost
  * LightGBM
  * Voting/Stacking Classifier
* Explainable AI techniques:

  * SHAP
  * LIME
  * Feature importance visualization
* Counterfactual explanation generation using:

  * DiCE
  * Alibi
  * CARLA

The system should:

* Provide accurate PCOD predictions
* Explain prediction reasoning
* Suggest minimal feature changes required to alter predictions
* Include proper visualizations and evaluation metrics

---

Some alternative approaches considered:

* Using a single ML model instead of ensemble learning
* Traditional feature importance methods without XAI frameworks
* Rule-based medical prediction systems
* Deep learning-only approaches without interpretability

However, ensemble learning combined with XAI and counterfactual explanations provides better robustness, transparency, and usability for healthcare-related AI systems.

---

**Additional context**
Suggested tech stack:

* Python
* Scikit-learn
* XGBoost / LightGBM / CatBoost
* SHAP
* LIME
* DiCE / Alibi

Possible deliverables:

* Modular codebase
* Jupyter notebook/demo
* Model evaluation report
* Visualization dashboard
* README documentation

This feature would be highly valuable for contributors interested in:

* Healthcare AI
* Responsible AI
* Explainable Machine Learning
* Research-oriented open-source contributions

My GSSoC id @Pritam27112004

### LINKED ISSUE #1622: PCOD Detection
## 🩺 PCOD Risk Detection System

A machine learning web application to predict PCOD (Polycystic Ovary Disease) risk 
from menstrual health data, with full Explainable AI integration.

# Features
- XGBoost Classifier** for risk prediction (binary classification)
- SHAP (Shapley Additive explanations) for global feature importance
- LIME (Local Interpretable Model-agnostic Explanations) for per-prediction explanation
- Real-time interactive UI built with **Streamlit**
- Sidebar sliders for patient data input
- Clinical recommendations based on prediction probability

# Dataset
Menstrual Health and PCOD Risk Detection Dataset (Kaggle)  
Features: BMI, cycle length, menses score, age, weight, height, etc.

# Tech Stack
- Python, Streamlit, XGBoost, SHAP, LIME, scikit-learn, pandas, matplotlib

# How to Run
```bash
pip install streamlit xgboost shap lime scikit-learn pandas matplotlib
streamlit run app.py

### PR DESCRIPTION
**Pull request by Shashank1725**
### What does this PR add?
A complete PCOD/PCOS Detection project using Machine Learning with explainability.

### Features:
- ✅ Random Forest & XGBoost models
- ✅ SHAP explainability (shap_summary.png)
- ✅ Streamlit web app (app.py)
- ✅ Complete Jupyter Notebook analysis (PCOD_Detection_Analysis.ipynb)
- ✅ Proper README.md documentation

### Dataset:
Kaggle PCOS Dataset

### Tech Stack:
Python, Scikit-learn, XGBoost, SHAP, Streamlit, Pandas, NumPy

### CODE REVIEW

File: PCOD Detection/lime_explanation.html
Comment (Sibam-Paul) [inline PCOD Detection/lime_explanation.html:1]: what is this for ?

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
1. remove the pycache
2. .vscode remove it if not linked with the codebase
@Shashank1725

**Comment by Sibam-Paul** [comment]
@Shashank1725 make the changes

**Review by Shashank1725 — COMMENTED** [review]
Deleted .vscode

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
Image

delete the pycache

File: PCOD Detection/PCOD_Detection_Analysis.ipynb
Comment (Sibam-Paul) [inline PCOD Detection/PCOD_Detection_Analysis.ipynb:1]: Image

follow the format of pr #1952 
link the csv using gdown or other way 

write the readme as the pr format provided 

for images make a pdf and explain in details

see the pinned issues for guidelines 

@Shashank1725 @Shashank1725

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
@Shashank1725 make the changes please

File: PCOD Detection/period - Copy.csv
Comment (Sibam-Paul) [inline PCOD Detection/period - Copy.csv:1]: remove the csv file add it as a link @Shashank1725 
see pr #1592 how it is done

File: PCOD Detection/lime_explanation.html
Comment (Shashank1725) [reply PCOD Detection/lime_explanation.html:1]: The purpose of LIME.html is to provide a visual explanation of the model's prediction for an individual patient, showing which features contributed most to the prediction.

File: PCOD Detection/README.md
Comment (Sibam-Paul) [inline PCOD Detection/README.md:1]: add info about dataset

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
